netboot-services/ipxeMenuGenerator/main.go

```diff
@@ -33,10 +33,39 @@ func main() {
 		if netbootServerIP == "" {
 			log.Fatal("NETBOOT_SERVER_IP not set")
 		}
+		azureNetbootServerIP := os.Getenv("AZURE_NETBOOT_SERVER_IP")
+		if azureNetbootServerIP == "" {
+			log.Fatal("AZURE_NETBOOT_SERVER_IP not set")
+		}
+
+		onpremExposedNetbootServer := os.Getenv("ONPREM_NETBOOT_SERVER")
+		if onpremExposedNetbootServer == "" {
+			log.Fatal("ONPREM_NETBOOT_SERVER not set")
+		}
+
+		azureBlobstorageSASToken := os.Getenv("AZURE_SYNC_SAS_TOKEN")
```

Comment:
Is the different behavior intended between `AZURE_SYNC_SAS_TOKEN` / `AZURE_SYNC_BLOB_URL` not set and the other environment variables. (`log.Error()` vs `log.Fatal()`)

Reply:
I wasn't sure if I should fatal here, if no azure details have been provided, as azure is the fallback option. I'm ok with it if we do check every variable to be filled out.

Reply:
Moved to fatal